src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -605,6 +605,53 @@ public SolutionState RemoveProject(ProjectId projectId)
             return builder.ToImmutable();
         }
 
+        private ImmutableDictionary<string, ImmutableArray<DocumentId>> CreateFilePathToDocumentIdsMapWithRenamedDocuments(TextDocumentState originalState, TextDocumentState newState)
+        => CreateFilePathToDocumentIdsMapWithRenamedDocuments(SpecializedCollections.SingletonEnumerable((originalState, newState)));
+
+        private ImmutableDictionary<string, ImmutableArray<DocumentId>> CreateFilePathToDocumentIdsMapWithRenamedDocuments(IEnumerable<(TextDocumentState originalState, TextDocumentState newState)> documentStates)
```

Comment:
Can we just implement this as remove + add since we already have those helpers? The overhead isn't much more, but the duplicate code is icky here.